master/pkg/model/experiment.go

```diff
@@ -517,6 +519,24 @@ func (t TrialLogBatch) ForEach(f func(interface{}) error) error {
 	return nil
 }
 
+// TrialProfilerMetricsBatchBatch represents a batch of trialv1.TrialProfilerMetricsBatch.
+type TrialProfilerMetricsBatchBatch []*trialv1.TrialProfilerMetricsBatch
```

Comment:
generics, gah! this whole batch this is just because i need to len(T[]) a few places. In 1.18 i'll make this beautiful.